backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/util/AsyncAggregatingSubscriber.java

```diff
@@ -59,18 +60,25 @@ public void onNext(final CompletableFuture<T> future) {
           if (throwable == null) {
             LOG.trace("Completed: {}", result);
             results.add(result);
+            phaser.arrive();
+            if (!phaser.isTerminated()) {
```

Comment:
:wrench: `phaser.arrive()` already returns a negative value if the phaser is terminated